processor/src/main/java/org/mapstruct/ap/internal/model/assignment/AdderWrapper.java

```diff
@@ -44,6 +48,7 @@ public AdderWrapper( Assignment rhs,
         this.thrownTypesToExclude = thrownTypesToExclude;
         String desiredName = Nouns.singularize( targetPropertyName );
         rhs.setSourceLocalVarName( rhs.createLocalVarName( desiredName ) );
+        adderType = first( getSourceType().determineTypeArguments( Collection.class ) );
```

Comment:
and `first` yields null if there is no type-bound..? Think that should be a comment here.